pinot-spi/src/main/java/org/apache/pinot/spi/secretstore/SecretStoreException.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.secretstore;
+
+/**
+ * Exception thrown when operations on the {@link SecretStore} fail.
+ * This exception encapsulates errors that may occur during secret storage,
+ * retrieval, updating, or deletion operations.
+ */
+public class SecretStoreException extends RuntimeException {
```

Comment:
Instead of catching all errors under SecretStoreException, you might want to add errorCode and a few exceptions to extend:
	•	SecretNotFoundException
	•	SecretPermissionException
	•	SecretStoreConnectionException

Reply:
Would it be okay to start with this exception type and just add error codes to categorize the common failure patterns? We can extend this class later if needed. I am adding error codes for now to enhance this exception class.

Reply:
As discussed offline, I have added static exception types for common failure scenarios.